config/rk-kotlin.el

```diff
@@ -0,0 +1,20 @@
+;;; rk-kotlin.el --- Configuration for kotlin.  -*- lexical-binding: t; -*-
+
+;;; Commentary:
+
+;;; Code:
+(eval-when-compile
+  (require 'use-package))
+
+(use-package kotlin-mode
+  :straight t
+  :defer t
+  :config
+  (progn
+    (add-to-list 'exec-path "~/kotlin-language-server/server/build/install/server/bin"))
```

Comment:
You're better off customizing this with `lsp-kotlin-language-server-path` rather than adding this to exec-path. I'd also create a fn that customises & enables the LSP for Kotlin and use that in the hook.